### ISSUE TITLE
ui-shared: Accessibility of Switch control

### ISSUE THREAD

**Issue by DataTriny** [keycloak/keycloak#42403]
Relates to #36480

### Before reporting an issue

- [x] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

admin/ui

### Describe the bug

Label for the Switch control is unhelpful for screen reader users. When using tab navigation, only "On" or "Off" is reported instead of the label that visually appear on the left of the control. Furthermore, this is redundant as switches are reported as checked or unchecked checkboxes. Screen reader users can still navigate the document and find the label that is before the switch, but this becomes confusing as one must remember that the label comes before the switch. At least the `aria-label` attribute on the control should be set to the same value as the text on the left to override the useless "On"/"Off" value.

Also worth noting that the way this control is built is considered bad practice by the developers of the underlying library:

> To keep inline with accessibility guidelines, the label for a switch must never dynamically change. A dynamically changing label (such as one label for an "on" state and another label for an "off" state) can be confusing as the contexts for each label changes as the switch state does. This applies to both visible text labels as well as labels provided via aria-label.

[patternfly Switch documentation](https://www.patternfly.org/components/switch/)

### Version

main branch

### Regression

- [ ] The issue is a regression

### Expected behavior

Label for switches must be the same for screen reader users as it is for everyone else.

### Actual behavior

When using tab navigation with a screen reader, only "On" or "Off" is reported instead of the label that visually appear on the left of the control.

### How to Reproduce?

With a screen reader on, tab into any switch such as the one labelled "Enabled" on the "Create realm" dialog, observe that the label that gets read is "On" instead of "Enabled".
